src/cs/Ssh/MultiChannelStream.cs

```diff
@@ -304,6 +287,15 @@ public async Task ConnectAndRunUntilClosedAsync(CancellationToken cancellation =
 		return channel;
 	}
 
+	/// <summary>
+	/// Creates a stream instance for a channel. May be overridden to create a
+	/// <see cref="SshStream" /> subclass.
+	/// </summary>
+	protected virtual SshStream CreateStream(SshChannel channel)
```

Comment:
It's used in async methods and can also be async

Reply:
I can't think of any reason it would be useful to make it async.